### PR TITLE
Use Pango for text layout.

### PR DESCRIPTION
**Pull request by asmaAL-Bahanta**
Use Pango for text layout to support languages that require complex text layout such as Arabic, Hebrew, and Indic language.The cairo_show_text() is part of cairo toy text API and should not be used in serious applications, see: http://cairographics.org/manual/cairo-text.html#cairo-show-text

### CODE REVIEW

**Comment by asmaAL-Bahanta** [comment]
This pull request is  for the other files where they use cairo toy text API.

File: src/bauhaus/bauhaus.c [hunk 1]
```diff
@@ -1578,18 +1578,24 @@ static gboolean dt_bauhaus_popup_draw(GtkWidget *widget, cairo_t *crf, gpointer
   if(darktable.bauhaus->keys_cnt)
   {
     cairo_save(cr);
-    cairo_text_extents_t ext;
-    set_text_color(cr, 1);
-    set_value_font(cr);
+	PangoLayout *layout;
```
Comment (LebedevRI) [inline src/bauhaus/bauhaus.c]: Wrong formatting

File: src/bauhaus/bauhaus.c [hunk 2]
```diff
@@ -1578,18 +1578,24 @@ static gboolean dt_bauhaus_popup_draw(GtkWidget *widget, cairo_t *crf, gpointer
...
+	PangoRectangle ink;
+	PangoFontDescription *desc = pango_font_description_from_string("sans-serif bold");
```
Comment (houz) [inline src/bauhaus/bauhaus.c]: `pango_font_description_free(desc)` and `g_object_unref(layout)`.

**Comment by asmaAL-Bahanta** [comment]
I have addressed all the comments. I hope you can test my patch and see if there is an issue especially in text positions in the GUI.

**Comment by houz** [comment]
I just noticed some small problems with the way you handle pango. As they are also present in the last PR I will just clean it up and merge this afterwards.

**Pull request merged**